device/api/umd/device/tt_soc_descriptor.h

```diff
@@ -149,14 +151,18 @@ class tt_SocDescriptor {
     void get_cores_and_grid_size_from_coordinate_manager();
 
     static tt_xy_pair calculate_grid_size(const std::vector<tt_xy_pair> &cores);
+    void translate_coordinates(
+        std::map<std::pair<CoreType, CoordSystem>, std::vector<tt::umd::CoreCoord>> &core_coord_map,
+        const CoreType core_type,
+        const CoordSystem coord_system) const;
 
     // TODO: change this to unique pointer as soon as copying of tt_SocDescriptor
     // is not needed anymore. Soc descriptor and coordinate manager should be
     // created once per chip.
     std::shared_ptr<CoordinateManager> coordinate_manager = nullptr;
-    std::map<CoreType, std::vector<tt::umd::CoreCoord>> cores_map;
+    mutable std::map<std::pair<CoreType, CoordSystem>, std::vector<tt::umd::CoreCoord>> cores_map;
```

Comment:
Do we actually need complication with lazy buffer and "double" buffering in soc descriptor, since we have translation map in coordinate_manager already?

Reply:
Removed lazy buffering